### PR TITLE
Update stale reference to etcd modifiedIndex

### PR DESCRIPTION
**Pull request by porridge**

### CODE REVIEW

**Comment by porridge** [comment]
/remove-lifecycle stale
I'm still waiting for a review 😢

File: contributors/devel/sig-architecture/api-conventions.md [hunk 1]
```diff
@@ -791,7 +791,7 @@ read/modify/write cycle, by verifying that the current value of resourceVersion
 matches the specified value.
 
 The resourceVersion is currently backed by [etcd's
-modifiedIndex](https://coreos.com/etcd/docs/latest/v2/api.html).
+mod_revision](https://etcd.io/docs/latest/learning/api/#key-value-pair).
```
Comment (krmayankk) [inline contributors/devel/sig-architecture/api-conventions.md:794]: @porridge  is this mainly due to migration from etcdv2 to etcd v3 ?

Comment (porridge) [reply contributors/devel/sig-architecture/api-conventions.md:794]: It's mostly due to the old URL disappearing
Screenshot from 2022-11-04 08-41-57

**Pull request merged**